schema_based_on_diagram.sql

```diff
@@ -0,0 +1,41 @@
+CREATE TABLE patients (
+  id INT GENERATED ALWAYS AS IDENTITY,
+  name VARCHAR(255),
+  date_of_birth DATE,
+  PRIMARY KEY(id)
+);
+
+
+CREATE TABLE medical_histories (
+  id INT GENERATED ALWAYS AS IDENTITY,
+  admitted_at TIMESTAMP,
+  patient_id INT REFERENCES patients(id),
+  status VARCHAR(100),
+  PRIMARY KEY(id)
+);
+
+CREATE TABLE invoices (
+  id INT GENERATED ALWAYS AS IDENTITY,
+  total_amount DECIMAL,
+  generated_at TIMESTAMP,
+  payed_at TIMESTAMP,
+  medical_history_id INT REFERENCES medical_histories(id),
+  PRIMARY KEY(id)
+);
+
+CREATE TABLE invoice_item (
+  id INT GENERATED ALWAYS AS IDENTITY,
+  unit_price DECIMAL,
+  quantity INT,
+  total_price DECIMAL,
+  invoice_id INT REFERENCES invoices(id),
+  treatment_id INT REFERENCES treatments(id),
+  PRIMARY KEY(id)
+);
+
+CREATE TABLE treatments (
+  id INT GENERATED ALWAYS AS IDENTITY,
+  type VARCHAR(255),
+  name VARCHAR(255),
+  PRIMARY KEY(id)
+);
```

Comment:
- [ ] It is a requirement that all FOREIGN KEYS have indexes created according to how you named your tables. Checkout the code snippet below.
![image](https://user-images.githubusercontent.com/81932478/214355728-c75fcf43-a435-458b-b300-7153a6e2deb9.png)